Add regression test for `echo 'mod unknown;' | rustc -`

Enselic opened the pull request:
Closes #65601

The bug is fixed since long ago, probably by #69838 (see https://github.com/rust-lang/rust/issues/65601#issuecomment-1650508071 for more details).
Add a regression test so we can close the issue.

Enselic committed:
Add regression test for `echo 'mod unknown;' | rustc -`

Mark-Simulacrum commented:
@bors r+ rollup

bors commented:
:pushpin: Commit 86433e6ed1f28d1961d8316022277287330d71d1 has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum 86433e6ed1f28d1961d8316022277287330d71d1 -->
<!-- homu: {"type":"Approved","sha":"86433e6ed1f28d1961d8316022277287330d71d1","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

matthiaskrgr commented:
@bors r-
https://github.com/rust-lang/rust/pull/114295#issuecomment-1659136869

Enselic committed:
tests/run-make/unknown-mod-stdin: Ignore Windows

The test fails on Windows with the diff

    -  = help: to create the module `unknown`, create file "unknown.rs" or "unknown/mod.rs"
    +  = help: to create the module `unknown`, create file "unknown.rs" or "unknown\mod.rs"

There is no need to run this test on Windows, so ignore it.

Enselic commented:
The test fails because of this diff:

```
-  = help: to create the module `unknown`, create file "unknown.rs" or "unknown/mod.rs"
+  = help: to create the module `unknown`, create file "unknown.rs" or "unknown\mod.rs"
```

There is no need to run this test on Windows, so let's just not. I have now added `# ignore-windows`. (There is plenty of prior art: `git grep '# ignore-windows' | wc -l` counts to 37 Makefiles.)

@rustbot ready

Mark-Simulacrum commented:
@bors r+

bors commented:
:pushpin: Commit 392bfc6bd34f49b51f62bd2788857610cc629dee has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum 392bfc6bd34f49b51f62bd2788857610cc629dee -->
<!-- homu: {"type":"Approved","sha":"392bfc6bd34f49b51f62bd2788857610cc629dee","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
